### PR TITLE
improve documentation (fix #745)

### LINKED ISSUE #745: Unclear documentation and typos
In several `.md` and `.rst` files, there is inconsistent language and typos (e.g. "setup" instead of "set up")

### PR DESCRIPTION
**Pull request by spencerng**
### Description

I fixed typos and grammatical issues in the `CONTRIBUTING.md` and `README.md` files and improved the formatting and style (to ensure consistency) of the documentation `.rst` files.

What component does this PR affect? 

- [ ] Server
- [ ] Console
- [ ] CLI
- [x] Docs
- [ ] Community Content
- [ ] Build System

Requires changes from other components? If yes, please mark the components:

- [ ] Server
- [ ] Console
- [ ] CLI
- [ ] Docs
- [ ] Community Content
- [ ] Build System

### Related Issue

#745 

### Solution and Design

I read over the README and CONTRIBUTING Markdown files, then fixed errors as I saw them.

### Type

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Docs update
- [ ] Community content

### Checklist:

- [x] I have read the **[contributing guide](https://github.com/hasura/graphql-engine/blob/master/CONTRIBUTING.md)** and my code conforms to the guidelines.
- [ ] This change requires a change in the documentation. 
- [x] I have updated the documentation accordingly.
- [ ] I have added required tests.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -69,15 +69,16 @@ before-mentioned sections).
 
 ## Commit messages
 
- - The first line should be a summary of the changes - not exceeding 50
-   characters. Followed by an optional body which has more details about the
-   changes. (https://github.com/erlang/otp/wiki/writing-good-commit-messages)
+- The first line should be a summary of the changes, not exceeding 50
+  character, followed by an optional body which has more details about the
```
Comment (dsandip) [inline CONTRIBUTING.md]: Change `character` to `characters`

File: server/CONTRIBUTING.md [hunk 2]
```diff
@@ -1,7 +1,7 @@
 # Contributing
 
-This guide is for setting-up the graphql-engine server for development on your
-own machine, and how to contribute.
+This guide is explains how to set up the graphql-engine server for development on your
```
Comment (dsandip) [inline server/CONTRIBUTING.md]: Should be "This guide explains how to set up.."

**Comment by spencerng** [comment]
@dsandip I've made the necessary changes. Sorry for the careless typos!

**Comment by dsandip** [comment]
@rikinsk over to you!

**Pull request merged**